packages/common-grpc/src/service.js

```diff
@@ -168,6 +176,17 @@ function GrpcService(config, options) {
     return global.GCLOUD_SANDBOX_ENV;
   }
 
+  if (config.defaultApiEndpoint) {
```

Comment:
Ideally, this should all be done in `Service`, but let me know I'm overlooking something.

Reply:
Looking at it, I'd stop relying on config.{baseUrl, customEndpoint} in the GrpcService constructor and use this.{baseUrl, customEndpoint} since it's being populated by Service. I'll make those changes if you're ok with them

Reply:
I think that should be fine, basically leaving as much work as possible to `Service`.